oak-blob-composite/src/main/java/org/apache/jackrabbit/oak/blob/composite/DelegateMinRecordLengthSelector.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.jackrabbit.oak.blob.composite;
+
+public interface DelegateMinRecordLengthSelector {
```

Comment:
Maybe not, I don't know.

It's hard to know what minRecordLength even means with CompositeDataStore.  It is unclear to me what this is actually used for within Oak itself.

If Oak actually calls (or if it were to call, in the future) the `minRecordLength()` method of the registered data store in order to determine if it should try to write a blob to the data store or to the node store, then it definitely matters.  In fact it becomes somewhat of a big mess with a `CompositeDataStore` because whether a blob can be written to the `CompositeDataStore` depends entirely on what delegate the blob would be written to and what the value of `minRecordLength` is for that delegate.

For the currently implemented use case for `CompositeDataStore` it doesn't matter but it could in the future if other supported use cases are added.  Suppose we were to support a scenario where a `CompositeDataStore` could have two delegates, both writable, one for everything under the `/images` path and another delegate for everything else.  Suppose the delegate containing the images is configured with `minRecordLength=16384` and the other delegate containing everything else is configured with `minRecordLength=8192`.  Now suppose a binary being written is of size 12000.  What value of `minRecordLength` should the composite data store return?

Since I wasn't sure which approach was best I made it an interface so we can provide different strategies.  If you don't think it is necessary we can eliminate it.

Reply:
{{minRecordLength}} decides upto what length the data can be encoded itself in the ID returned with no actual blob being stored in the datastore.
With the {{CompositeDataStore}} the problem is that it is decorated by the DataStoreBlobStore and thus needs *a* value. If the encoding logic was in the {{CompositeDataStore}} it would have been easier to have the individual delegates take that decision.
With this context in mind I don't think the there is scope for any other value than the minimum and thus eliminate the strategies. We should simply add that in the documentation that the {{minRecordLength}} should be configured the same otherwise the min value takes affect.

Reply:
Ok - I'll move it just to `CompositeDataStore` then, that will be much simpler.